sklearn/svm/tests/test_sparse.py

```diff
@@ -135,20 +136,77 @@ def kfunc(x, y):
     assert_array_equal(clf_lin.predict(X_sp), clf_mylin.predict(X_sp))
 
 
-def test_svc_iris():
-    # Test the sparse SVC with the iris dataset
-    for k in ('linear', 'poly', 'rbf'):
-        sp_clf = svm.SVC(gamma='scale', kernel=k).fit(iris.data, iris.target)
-        clf = svm.SVC(gamma='scale', kernel=k).fit(iris.data.toarray(),
-                                                   iris.target)
-
-        assert_array_almost_equal(clf.support_vectors_,
-                                  sp_clf.support_vectors_.toarray())
-        assert_array_almost_equal(clf.dual_coef_, sp_clf.dual_coef_.toarray())
-        assert_array_almost_equal(
-            clf.predict(iris.data.toarray()), sp_clf.predict(iris.data))
-        if k == 'linear':
-            assert_array_almost_equal(clf.coef_, sp_clf.coef_.toarray())
+def _toarray(a):
+    if sparse.issparse(a):
+        return a.toarray()
+    return a
+
+
+def _assert_svc_equal(svc1, svc2, atol=1e-7):
+    # Check that 2 support vector machines parametrize the equivalent decision
+    # functions by comparing dual coefficients and support vectors.
+    assert svc1.get_params() == svc2.get_params()
+
+    sv1 = _toarray(svc1.support_vectors_)
+    dc1 = _toarray(svc1.dual_coef_)
+    sv2 = _toarray(svc2.support_vectors_)
+    dc2 = _toarray(svc2.dual_coef_)
+    assert dc1.shape == dc2.shape
```

Comment:
Actually sets of support vectors may be different. If the input contains duplicates samples, and that point happens to be a support vector, then `support_vectors_` may contains a single, or multiple entries depending on history of accumulation of floating point errors.